SPHINXsys/src/shared/particle_dynamics/electro_physiology/electro_physiology.h

```diff
@@ -92,52 +92,55 @@ namespace SPH
 		virtual ~AlievPanfilowModel(){};
 	};
 
+	// type trait for pass type template constructor
+	template <typename T>
+	struct encodeType
```

Comment:
The regular name for this `type_identity`, see for example standardized version from C++20 https://en.cppreference.com/w/cpp/types/type_identity

Reply:
Here, I used it because the conflict between the parameter of a template class to that of a template constructor.
Currently, I guess C++20 is not fully supported yet.

Reply:
I have modify the code as you suggetsed.